src/providers/data_provider_be.c

```diff
@@ -249,7 +249,7 @@ static errno_t be_check_online_request(struct be_ctx *be_ctx)
 static void be_check_online_done(struct tevent_req *req)
 {
     struct be_ctx *be_ctx;
-    struct dp_reply_std reply;
```

Comment:
I guess we never caught this bug because of the (void**) cast, but I wonder how did you find this bug? Did it cause any issues in the code? If yes,it might be a good candidate for a backport.

Reply:
I found this bug while debugging why the "Backend is Online" and "Backend is Offline" messages were never triggered. Apart from this small issue, I'd say reset_fo() was being called way more times than actually needed.

So, yes, probably this patch could be backported.